pkg/controller/astarte/reconcile/astarte_generic_api.go

```diff
@@ -224,15 +219,34 @@ func getAstarteGenericAPIEnvVars(deploymentName string, cr *apiv1alpha1.Astarte,
 		// Add Cassandra Nodes, AMQP information and Max results count
 		rabbitMQHost, rabbitMQPort := misc.GetRabbitMQHostnameAndPort(cr)
 		userCredentialsSecretName, userCredentialsSecretUsernameKey, userCredentialsSecretPasswordKey := misc.GetRabbitMQUserCredentialsSecret(cr)
+
+		cassandraPrefix := ""
+		v := getSemanticVersionForAstarteComponent(cr, cr.Spec.Components.AppengineAPI.Version)
+		checkVersion, _ := v.SetPrerelease("")
+		constraint, _ := semver.NewConstraint("< 1.0.0")
+		if constraint.Check(&checkVersion) {
+			cassandraPrefix = "ASTARTE_"
+		}
+
+		// Add Cassandra Nodes
+		ret = append(ret, v1.EnvVar{
+			Name:  cassandraPrefix + "CASSANDRA_NODES",
+			Value: getCassandraNodes(cr),
+		})
+
 		ret = append(ret,
 			v1.EnvVar{
-				Name:  "ASTARTE_CASSANDRA_NODES",
+				Name:  cassandraPrefix + "CASSANDRA_NODES",
 				Value: getCassandraNodes(cr),
 			},
 			v1.EnvVar{
 				Name:  "APPENGINE_API_MAX_RESULTS_LIMIT",
 				Value: strconv.Itoa(getAppEngineAPIMaxResultslimit(cr)),
 			},
+			v1.EnvVar{
+				Name:  "APPENGINE_API_ENABLE_COMPRESSION",
+				Value: strconv.FormatBool(pointy.BoolValue(cr.Spec.Components.AppengineAPI.EnableCompression, false)),
+			},
```

Comment:
This is a leftover from old Phoenix versions, @matt-mazzucato remove this from AppEngine in a fixup